### PR TITLE
Add istio support to gke-public-cluster

### PR DESCRIPTION
**Pull request by n0npax**
Ref: https://github.com/gruntwork-io/terraform-google-gke/issues/38

Add support for istio addon

### CODE REVIEW

**Comment by n0npax** [comment]
Requested changes:
* [ ] Add a README update describing the minimal setup needed to get started with Istio?
* [ ] Update the variable block to say EXPERIMENTAL FEATURES and mention that these are not recommended to be enable unless you know what you are doing?
* [ ] Add a regression test that verifies Istio setup?

File: examples/gke-public-cluster-istio/main.tf [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  # source = "github.com/gruntwork-io/terraform-google-gke.git//modules/gke-cluster?ref=v0.2.0"
+  source = "../../modules/gke-cluster"
+
+  name = var.cluster_name
+
+  project  = var.project
+  location = var.location
+
+  # enable istio
+  enable_istio = var.enable_istio
```
Comment (yorinasub17) [inline examples/gke-public-cluster-istio/main.tf]: You can probably hard code this to `true` for this example.

**Comment by yorinasub17** [comment]
Also would like to hear what @robmorgan and @autero1 thinks.

File: modules/gke-cluster/main.tf [hunk 2]
```diff
@@ -88,6 +88,12 @@ resource "google_container_cluster" "cluster" {
     network_policy_config {
       disabled = ! var.enable_network_policy
     }
+
+    istio_config {
+      disabled = ! var.enable_istio
+      auth = var.istio_auth
```
Comment (yorinasub17) [inline modules/gke-cluster/main.tf]: Can you run `terraform fmt`?

File: examples/gke-public-cluster-istio/main.tf (hunk 1 above)
Comment (n0npax) [reply examples/gke-public-cluster-istio/main.tf]: fixed + removed `enable_istio` from variables in istio example.

File: modules/gke-cluster/main.tf (hunk 2 above)
Comment (n0npax) [reply modules/gke-cluster/main.tf]: sorry about this. Fixed.

**Comment by n0npax** [comment]
I hope it's ok. Thanks for the review.

**Comment by yorinasub17** [comment]
It looks like the build failed. The problem appears to be that when Istio is turned on, none of the nodes actually get registered to the cluster.

File: modules/gke-cluster/main.tf (hunk 2 above)
Comment (robmorgan) [reply modules/gke-cluster/main.tf]: This block shouldn't have a newline underneath it either.

Comment (robmorgan) [reply modules/gke-cluster/main.tf]: Also I'm wondering if we should only add the `istio_config` block is the `enable_istio` parameter is set. I'm fairly certain we could adapter code like: https://github.com/gruntwork-io/terraform-google-gke/blob/74481db83eacfc047953bb1ec6f590e7ae93591f/modules/gke-cluster/main.tf#L47 to achieve this.

**Pull request closed without merging**